R/add_cell_mito_qc.R

```diff
@@ -35,15 +37,39 @@ add_cell_mito_qc <- function(sce, mito, ...){
     warning("sce does not contain genes corresponding to the list of mito gene names")
   }
 
+  # check that miQC is logical
+  if(!is.logical(miQC)){
+    stop("miQC must be set as TRUE or FALSE")
+  }
+
   # add per cell QC with mitochondrial subset
   sce <- scater::addPerCellQC(
     sce,
     subsets = list(mito = mito[mito %in% rownames(sce)]),
     ...
   )
-  # simplify naming
-  names(colData(sce)) <- stringr::str_replace(names(colData(sce)),
-                                              "^subsets_mito_",
-                                              "mito_")
+
+
+  if(miQC){
+    # generate linear mixture model of probability of cells being compromised
+    model <- miQC::mixtureModel(sce)
```

Comment:
For an sce object that was 35 MB, the model was 500 KB and you _could_ store it as metadata but you are right if we did anything to the sce object later then it wouldn't be as useful anymore. I am not as inclined to store it here unless you had a particular idea for why you wanted to store it?

Reply:
My main thought was that it would make it possible for somebody to come back and run `miQC::filterCells()` with other options if they so desired. Thinking about it, and looking at the code in https://github.com/greenelab/miQC/blob/main/R/filterCells.R, I actually think that even if the cells in the sce change, `miQC::filterCells()` will still work as expected. So I think including this may be worthwhile, partly because we may want to use it later to construct a suggested filter column.

Reply:
That makes sense. I went ahead and added the model into the metadata.